clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -142,9 +139,7 @@ private NetworkClient(MetadataUpdater metadataUpdater,
         this.correlation = 0;
         this.randOffset = new Random();
         this.requestTimeoutMs = requestTimeoutMs;
-        this.nodesEverSeen = new ArrayList<>();
-        this.nodesEverSeenById = new HashMap<>();
-        
+        this.bootstrapNodes = this.metadataUpdater.fetchNodes();
```

Comment:
I think the naming is a bit of a nit, although I agree initialNodes might be clearer since this won't be exactly the set of bootstrap nodes (it's retrieved indirectly and due to KAFKA-3112, it's possible some would be filtered out). 

But this is a little bit brittle as is. For this to work, it depends on the order in which the initial update is provided to `Metadata` since it is initialized with `Cluster.empty()`. I'm wondering if we should, as part of this patch, require the bootstrap `Cluster` to be included in the `Metadata constructor` to ensure we always have a list of bootstrap nodes.